pandas/core/indexes/datetimelike.py

```diff
@@ -773,7 +773,10 @@ def intersection(self, other, sort=False):
         else:
             lslice = slice(*left.slice_locs(start, end))
             left_chunk = left._values[lslice]
-            result = type(self)._simple_new(left_chunk)
+            # error: Argument 1 to "_simple_new" of "DatetimeIndexOpsMixin" has
+            # incompatible type "Union[ExtensionArray, Any]"; expected
+            # "Union[DatetimeArray, TimedeltaArray, PeriodArray]"  [arg-type]
+            result = type(self)._simple_new(left_chunk)  # type: ignore[arg-type]
```

Comment:
im fine with this, but i think we could just set `result = left_chunk`

Reply:
lets get ci green first, then discuss.